src/pages/word-duel.jsx

```diff
@@ -5,37 +5,97 @@ import Board from '../components/board';
 
 const MAX_GUESSES = 6;
 
+const EvaluationColor = {
+    Gray: "gray",
+    Yellow: "yellow",
+    Green: "green"
+}
+
 export default function WordDuel() {
     // const [ word, setWord ] = useState(generateNewWord())
-    const word = "awash";
-    const [ guesses, setGuesses ] = useState([]); 
-    const [ error, setError ] = useState("");
+    const word = "stink";
+    const [guesses, setGuesses] = useState([]);
+    const [evaluations, setEvaluations] = useState([]);
+    const [gameOver, setGameOver] = useState(false);
+    const [error, setError] = useState("");
+
     function checkGuess(guess) {
         if (guess.length !== word.length) {
             setError("Guess wrong length!");
             return false;
         }
 
+        if (guesses.length >= MAX_GUESSES) {
+            setError("Guess limit reached.");
+            setGameOver(true);
+            return false;
+        }
+        
         setError("");
-        setGuesses([ ...guesses, guess ])
+        setGuesses([...guesses, guess]);
+        evaluateGuess(guess);
         return true;
     }
 
+    function evaluateGuess(guess) {
+        let evaluation = [];
+        for (let i = 0; i < guess.length; i++) {
+            /* 
+            TODO: These are very rudamentary rule but they are sufficient for now 
+            There are some corner cases that can occur when there are repeated letters in a guess
+            Case 1: Guess contains multiples of a letter where only 1 is in the word and both "miss"
+                Actual: Both letters are yellow
+                Expected: One letter should be yellow (the first one) and the other should be gray
+            Case 2: Guess contains multiples of a letter where only 1 is in the word where 1 "hits" and 1 "misses"
+                Actual: "Miss" is yellow and "hit" is green
+                Expected: "Miss" is gray and "hit" is green
+            Need to devise a way to make sure each letter is only used once when evaluating a guess
+            */
+            if (guess.charAt(i) === word.charAt(i)) {
+                evaluation = [...evaluation, EvaluationColor.Green];
+                continue;
+            }
+
+            if (word.includes(guess.charAt(i))) {
+                evaluation = [...evaluation, EvaluationColor.Yellow];
+                continue;
+            }
+
+            evaluation = [...evaluation, EvaluationColor.Gray];
+        }
+
+        winGame(evaluation);
+
+        setEvaluations([...evaluations, evaluation]);
+    }
+
+    function winGame(evaluation) {
+        for (let i = 0; i < evaluation.length; i++) {
+            if(evaluation[i] !== "green") {
+                return;
+            }
+        }
+
+        setGameOver(true);
+        setError("You win!");
```

Comment:
Maybe we should update the `error` to be `messaging`.  Or, instead of overloading this for both errors and general game feedback, we could have a component that renders "YOU WIN" or "YOU LOSE" based on the state of the `gameOver` value

Reply:
Oh I agree with this for sure. I think just changing `error` to be `messaging` for now makes sense. Can add a component like the one described later on as a feature.